docs/concepts/serialization.md

```diff
@@ -465,6 +484,153 @@ annotated as `<SomeType>`, and type-checkers like mypy will treat the attribute
+which means that the model will ignore the schema and instead ask the object itself how it should be serialized.
+In particular, this means that when model subclasses are serialized, fields present in the subclass but not in
+the original schema will be included.
+
+If `serialize_as_any` is set to `False` (which is the default), the model will be serialized using the schema,
+which means that fields present in a subclass but not in the original schema will be ignored.
+
+For example:
+
+```py
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    name: str
+
+
+class UserLogin(User):
+    password: str
+
+
+class OuterModel(BaseModel):
+    user1: User
+    user2: User
+
+
+user = UserLogin(name='pydantic', password='password')
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=True)) # (1)!
+"""
+{
+    'user1': {'name': 'pydantic', 'password': 'password'},
+    'user2': {'name': 'pydantic', 'password': 'password'},
+}
+"""
+
+print(OuterModel(user1=user, user2=user).model_dump(serialize_as_any=False)) # (2)!
+"""
+{
+    'user1': {'name': 'pydantic'},
+    'user2': {'name': 'pydantic'},
+}
+"""
+```
+
+1. With `serialize_as_any` set to `True`, the result matches that of V1.
+2. With `serialize_as_any` set to `False` (the V2 default), fields present on the subclass,
+but not the base class, are not included in serialization.
+
+This setting even takes effect with nested and recursive patterns as well. For example:
+
+```py
+from typing import List
+
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    name: str
+    friends: List['User']
+
+class UserLogin(User):
+    password: str
+
+
+class OuterModel(BaseModel):
+    user: User
+
+user = UserLogin(name='samuel', password='pydantic-pw', friends=[UserLogin(name='sebastian', password='fastapi-pw', friends=[])])
+
+print(OuterModel(user=user).model_dump(serialize_as_any=True)) # (1)!
+"""
+{
+    'user': {
+        'name': 'samuel',
+        'friends': [
+            {
+                'name': 'sebastian',
+                'friends': [],
+                'password': 'fastapi-pw',
+            }
+        ],
+        'password': 'pydantic-pw',
+    }
+}
+"""
+
+print(OuterModel(user=user).model_dump(serialize_as_any=False)) # (2)!
+"""
+{
+    'user': {
+        'name': 'samuel',
+        'friends': [
+            {
+                'name': 'sebastian',
+                'friends': [],
+            }
+        ],
+    }
+}
+"""
+```
+
+1. Even nested `User` model instances are dumped with fields unique to `User` subclasses.
+2. Even nested `User` model instances are dumped without fields unique to `User` subclasses.
+
+#### Overridng the `serialize_as_any` default (False)
+
+You can override the default setting for `serialize_as_any` at the project level by configuring a subclass of `BaseModel` that overrides the default for the `serialize_as_any` argument to `model_dump()` and `model_dump_json()`.
```

Comment:
I think it's confusing that you say "at the project level" here, but then have `class User(BaseModel):` below, where it would not use the serialize as any behavior by default.

I would suggest changing the phrasing here to something like:
```suggestion
You can override the default setting for `serialize_as_any` by configuring a subclass of `BaseModel` that overrides the default for the `serialize_as_any` argument to `model_dump()` and `model_dump_json()`, and then use that as the base class (instead of `pydantic.BaseModel`) for any model you want to have this default behavior.
```